### ISSUE TITLE
Multi-GPU training does not compute loss/metrics correctly

### ISSUE THREAD

**Issue by bermeitinger-b** [keras-team/keras#19340]
I'm training a model on multiple GPUs. During training, the progress bar does not show the average of each GPU but the sum, leading to e.g. for MNIST the accuracy for 4 GPUs is around 3.5734 (actual value is 0.89) and losses way off the validation loss.

After each epoch, the loss shows `nan`.

You can see this in this Colab: https://colab.research.google.com/drive/1Sp2fURPtSAD7VLGxB2GitI-bYb_XwY9h?usp=sharing

(Keras 3.1.1, Tensorflow 2.16.1)

This also happens on actual 4 GPUs, not only with this GPU-slicing on Colab as taken from one of the Keras' example notebooks.

```
Epoch 1/3
469/469 ━━━━━━━━━━━━━━━━━━━━ 28s 45ms/step - loss: nan - sparse_categorical_accuracy: 2.1701 - val_loss: 0.3587 - val_sparse_categorical_accuracy: 0.8993
Epoch 2/3
469/469 ━━━━━━━━━━━━━━━━━━━━ 14s 30ms/step - loss: nan - sparse_categorical_accuracy: 3.5678 - val_loss: 0.2637 - val_sparse_categorical_accuracy: 0.9236
Epoch 3/3
469/469 ━━━━━━━━━━━━━━━━━━━━ 13s 28ms/step - loss: nan - sparse_categorical_accuracy: 3.6713 - val_loss: 0.2188 - val_sparse_categorical_accuracy: 0.9339
```

Can I trust that the model is trained correctly?

**Comment by SuryanarayanaY** [comment]
Hi @bermeitinger-b ,

Thanks for reporting the issue. It seems this is duplicate of #19280 where Dev team looking into the issue. Thanks!

**Comment by fchollet** [comment]
The original issue has been fixed. Thanks for reporting.
